plugins/@grouparoo/vero/src/lib/export/exportRecord.ts

```diff
@@ -0,0 +1,107 @@
+import { Errors, ExportRecordPluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import { VeroClient } from "../client";
+import { handleGroupChanges } from "./listMethods";
+
+export const exportRecord = async (args) => {
+  try {
+    return await handlePersonChanges(args);
+  } catch (error) {
+    // look for the rate limit exceeded status code.
+    if (error?.response?.status === 429) {
+      const retryIn = Math.floor(Math.random() * 10) + 1;
+      return { error, success: false, retryDelay: 1000 * retryIn };
+    }
+    throw error;
+  }
+};
+
+const handlePersonChanges: ExportRecordPluginMethod = async ({
+  appOptions,
+  syncOperations,
+  export: {
+    newRecordProperties,
+    oldRecordProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newRecordProperties).length === 0) {
+    return { success: true };
+  }
+
+  if (!newRecordProperties["id"]) {
+    throw new Error(`newRecordProperties[id] is a required mapping`);
+  }
+
+  const client = await connect(appOptions);
+
+  const newID = newRecordProperties["id"];
+  const oldID = oldRecordProperties["id"];
+
+  if (oldID && oldID !== newID) {
```

Comment:
I wonder if this should be below the delete, so it doesn't change it.

Reply:
OI'm also curious how synchronous this is? What have you seen? For example, does it work as expected when it immediately gets updated or groups are updated.

Reply:
So let's move this down below the `toDelete` clause